blackbird_python/blackbird/tests/test_listener.py

```diff
@@ -336,16 +336,6 @@ def test_operation_mode_expressions(self, parse_input_mocked_metadata):
         )
         assert bb.operations == [{'op': 'MZgate', 'args': [0, 1], 'kwargs': {}, 'modes': [8, 3]}]
 
-    @pytest.mark.parametrize("arr", ["\t{phase_0}\n\t{phase_1}\n", "\t{phase_0}, {phase_1}\n"])
-    def test_parameter_idx(self, arr, parse_input_mocked_metadata):
-        """Test that parameter array indexing works inside arguments"""
```

Comment:
How come this test was deleted?

Reply:
Correct me if I'm wrong, but the way I understand it a BB script with free parameters (i.e. a template script) should never be run directly, and thus it's not necessarily possible to access free parameters before specific parameter values have been assigned, as is done in this test. 

Instead I guess we could add further tests that fill in the template and _then_ test whether things works the way they should.

Reply:
Got it! I didn't spot that the data in the parametrize decorator where templated parameters 🙂